src/Promotions/InPluginUpsells/resources/js/donation-options.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ *
+ * @unreleased
+ *
+ */
+declare global {
+    interface Window {
+        GiveLegacyFormEditor: GiveLegacyFormEditorData;
+    }
+}
+
+/**
+ *
+ * @unreleased
+ *
+ */
+export interface GiveLegacyFormEditorData {
+    apiRoot: string;
+    apiNonce: string;
+}
+
+/**
+ *
+ * @unreleased
+ *
+ */
+const recurringProductRecommendation = document.querySelector('.givewp-donation-options');
+const dismissAction = document.querySelector('.givewp-donation-options_close');
+
+function renderContainer() {
+    const preceedingContent = document.querySelector('._give_custom_amount_text_field');
+
+    preceedingContent.insertAdjacentElement('afterend', recurringProductRecommendation);
+}
+
+renderContainer();
```

Comment:
If `renderContainer()` is called directly after its definition then it can be refactored to procedural code - no need for the function definition.

Unless this function is used elsewhere that I missed?

Reply:
Resolved in favor of https://github.com/impress-org/givewp/pull/6802#discussion_r1195445746.